src/test/validation_block_tests.cpp

```diff
@@ -98,13 +96,12 @@ std::shared_ptr<CBlock> MinerTestingSetup::Block(const Config &config,
     // spend
     CMutableTransaction txCoinbase(*pblock->vtx[0]);
     if (nHeight > -1) {
-        txCoinbase.vin[0].scriptSig = CScript() << COINBASE_PREFIX << nHeight
-                                                << std::vector<uint8_t>(80);
+        txCoinbase.vin[0].scriptSig = CScript() << std::vector<uint8_t>(80);
     }
     txCoinbase.vout.resize(2);
     txCoinbase.vout[1].scriptPubKey = pubKey;
-    txCoinbase.vout[1].nValue = txCoinbase.vout[0].nValue;
-    txCoinbase.vout[0].nValue = Amount::zero();
+    txCoinbase.vout[0].scriptPubKey = CScript() << OP_RETURN << COINBASE_PREFIX
+                                                << nHeight << i++;
```

Comment:
The behavior of `i++` vs `++i` is really subtle in C++. It's preferable to put i++ on a separate line for clarity.

Reply:
Maybe include a comment as to what the i++ is for.